File: server/src/models/foods/FoodPreferenceModel.js
class FoodPreferenceModel {
  constructor({ id, name, value }) {
    this.id = id;
    this.name = name;
    this.value = value;
  }

  static fromJson(json) {
    return new FoodPreferenceModel({
      id: json["id"],
      name: json["name"],
      value: json["value"] == "1" ? true : json["value"] == "0" ? false : null,
    });
  }
}

export default FoodPreferenceModel;

File: lib/data/models/foods/food_preference_model.dart
import 'package:hive_flutter/hive_flutter.dart';

part 'food_preference_model.g.dart';

@HiveType(typeId: 6)
class FoodPreferenceModel extends HiveObject {
  FoodPreferenceModel({required this.id, required this.name, this.value});

  factory FoodPreferenceModel.fromJson(Map<String, dynamic> json) {
    return FoodPreferenceModel(
      id: json['id'] as String?,
      name: json['name'] as String?,
      value: json['value'] as bool?,
    );
  }

  @HiveField(0)
  final String? id;
  @HiveField(1)
  final String? name;
  @HiveField(2)
  final bool? value;
}
